### PR TITLE
Add ability to run script to generate images in docs

### LINKED ISSUE #23900: Allow MooseDocs to execute plot scripts
## Reason
This will allow plots to be generated that can be used in documentation, for instance, plots of assessment cases. This will support #21133, for example, as well as many applications.

## Design
See #23912 for the evolving design.

## Impact
New option.

### PR DESCRIPTION
**Pull request by aeslaughter**
closes #23900
## Reason

Add flexibility to documentation system to include the ability to run scripts that generate images for documentation using markdown.

## Design

The media extensions was expanded to include a `ScriptCommand` object that accepts a python script name. This script is automatically executed during the documentation build and the resulting image created by the script is managed within the documentation tools for rendering.

## Impact

This is a new feature and will not impact existing code/documentation.

### CODE REVIEW

**Comment by aeslaughter** [comment]
@joshuahansel @cticenhour This gets the script/image stuff working in parallel, I implemented it a bit different w/r/t to syntax so feel free to change it back. The pages list in the executioner cannot be modified b/c each page has attached a translator object which doesn't pickle. The per page translator allows for each page to be rendered differently (e.g., a presentation vs. a normal page). Thus, I used the page global meta data that is maintained by MooseDocs to provide the necessary data to the local processes while they are running.

Given the constraints of python for threads, this is probably about as clean as it will get.

**Comment by aeslaughter** [comment]
This also closes #23912 if merged

File: python/MooseDocs/extensions/media.py [hunk 1]
```diff
@@ -94,6 +111,52 @@ def createToken(self, parent, info, page, settings):
...
+
+        # Generate the plot
+        LOG.info("Executing plot script %s", script_path)
+        result = subprocess.run(["python", script_path], capture_output=True)
+        if result.returncode != 0:
+            msg = "Failed to execute python script: %s"
+            raise exceptions.MooseDocsException(msg, script_path)
+
+        # Currently the plot is assumed to reside in the same directory as the plot script.
+        plot_name = settings['image_name'] or os.path.basename(script_path).replace('.py', '.png')
```
Comment (loganharbour) [inline python/MooseDocs/extensions/media.py:141]: 🙌🏼

**Comment by aeslaughter** [comment]
Probably just need to check if it exists before the addPage method is called.

**Comment by joshuahansel** [comment]
@loganharbour @cticenhour Added doc, so this is ready for review!

**Review by cticenhour — APPROVED** [review]
Thank you for the news and doco @joshuahansel, and thanks again for your help on this @aeslaughter! 🍻

**Pull request merged**